### PR TITLE
[Remove Stream/Optional usage] LPS-172541 module: portal-search-tuning-synonyms-web

### PR DESCRIPTION
**Pull request by haydenzheng314**
Previous: https://github.com/liferay-core-infra/liferay-portal/pull/3354
Ticket: https://issues.liferay.com/browse/LPS-172541
Thanks

### CODE REVIEW

**Comment by LisztLiszt** [comment]
ci:stop

**Comment by LisztLiszt** [comment]
ci:test:sf

**Comment by LisztLiszt** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:test:search

**Comment by Kvale1733** [comment]
@LisztLiszt No regressions found.

**Comment by lipusz** [comment]
Ping QA Search to verify the search suite results.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommand.java [hunk 1]
```diff
@@ -88,19 +88,20 @@ protected void updateSynonymSet(ActionRequest actionRequest)
...
-		synonymSetOptional.ifPresent(
-			synonymSet -> synonymSetBuilder.synonymSetDocumentId(
-				synonymSet.getSynonymSetDocumentId()));
+		if (synonymSet != null) {
+			synonymSetBuilder.synonymSetDocumentId(
+				synonymSet.getSynonymSetDocumentId());
+		}
 
-		if (synonymSetOptional.isPresent()) {
+		if (synonymSet != null) {
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommand.java:104]: @LisztLiszt These 2 `if` are with same condition, please merge them.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImpl.java [hunk 2]
```diff
@@ -99,10 +100,10 @@ private Optional<Document> _getDocumentOptional(
 			getDocumentRequest);
 
 		if (getDocumentResponse.isExists()) {
-			return Optional.of(getDocumentResponse.getDocument());
+			return getDocumentResponse.getDocument();
 		}
 
-		return Optional.empty();
+		return null;
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImpl.java:106]: @LisztLiszt Let's make it a convention: `return null` in the shortcut. Please flip this if-shortcut.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/display/context/EditSynonymSetsDisplayBuilder.java [hunk 3]
```diff
@@ -86,23 +83,29 @@ private String _getRedirect() {
 		return ParamUtil.getString(_httpServletRequest, "redirect");
 	}
 
-	private Optional<SynonymSet> _getSynonymSetOptional(long companyId) {
-		SynonymSetIndexName synonymSetIndexName =
-			_synonymSetIndexNameBuilder.getSynonymSetIndexName(companyId);
+	private SynonymSet _getSynonymSet(long companyId) {
+		String id = ParamUtil.getString(_renderRequest, "synonymSetId", null);
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/display/context/EditSynonymSetsDisplayBuilder.java:87]: @LisztLiszt Please rename this variable to match the string in the method call. Basically, this is to get a parameter from a render request, so the variable name should be the parameter name `synonymSetId`.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommand.java [hunk 4]
```diff
@@ -58,14 +56,16 @@ protected void doProcessAction(
...
-	protected Optional<SynonymSet> getSynonymSetOptional(
+	protected SynonymSet getSynonymSet(
 		SynonymSetIndexName synonymSetIndexName, ActionRequest actionRequest) {
 
-		return Optional.ofNullable(
-			ParamUtil.getString(actionRequest, "synonymSetId", null)
-		).flatMap(
-			id -> _synonymSetIndexReader.fetchOptional(synonymSetIndexName, id)
-		);
+		String id = ParamUtil.getString(actionRequest, "synonymSetId", null);
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommand.java:62]: @LisztLiszt Also this variable.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommand.java (hunk 1 above)
Comment (dantewang) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommand.java:104]: @LisztLiszt After you merge the 2 `if`, there will be another issue. Usually we prefer `if-else` over `if_not-else`. So please flip the condition.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImplTest.java [hunk 5]
```diff
@@ -56,44 +55,42 @@ public void setUp() throws Exception {
...
-	public void testFetchOptionalWithDocExistsFalse() {
+	public void testFetchWithDocExistsFalse() {
 		setUpSearchEngineAdapter(_setUpGetDocumentResponse(false));
 
 		SynonymSetIndexName synonymSetIndexName = Mockito.mock(
 			SynonymSetIndexName.class);
 
-		Optional<SynonymSet> synonymSetOptional =
-			_synonymSetIndexReaderImpl.fetchOptional(synonymSetIndexName, "id");
+		SynonymSet synonymSet = _synonymSetIndexReaderImpl.fetch(
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImplTest.java:64]: @LisztLiszt Please inline variable. Before your change, a method of the optional is called, so we need the optional to be assigned to a variable. However, after the change, you are directly asserting the variable itself, so the variable is not needed.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImplTest.java [hunk 6]
```diff
@@ -56,44 +55,42 @@ public void setUp() throws Exception {
...
+		}
 	}
 
 	@Test
-	public void testFetchOptionalWithNullId() {
-		Optional<SynonymSet> synonymSetOptional =
-			_synonymSetIndexReaderImpl.fetchOptional(
-				Mockito.mock(SynonymSetIndexName.class), null);
+	public void testFetchWithNullId() {
+		SynonymSet synonymSet = _synonymSetIndexReaderImpl.fetch(
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImplTest.java:90]: @LisztLiszt Also this variable.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommandTest.java [hunk 7]
```diff
@@ -90,22 +88,18 @@ public void testDoProcessAction() throws Exception {
 	}
 
 	@Test
-	public void testGetSynonymSetOptional() {
-		Optional<SynonymSet> synonymSetOptional =
-			_editSynonymSetsMVCActionCommand.getSynonymSetOptional(
-				Mockito.mock(SynonymSetIndexName.class), _actionRequest);
+	public void testGetSynonymSet() {
+		SynonymSet synonymSet = _editSynonymSetsMVCActionCommand.getSynonymSet(
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommandTest.java:92]: @LisztLiszt Please inline this one. Declare the variable in line 101, as we need to call methods on that one.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommandTest.java [hunk 8]
```diff
@@ -124,11 +118,11 @@ public void testUpdateSynonymSet() throws Exception {
 
 	@Test
 	public void testUpdateSynonymSetIndex() throws PortalException {
-		Optional<SynonymSet> synonymSetOptional = Optional.empty();
+		SynonymSet synonymSet = null;
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/test/java/com/liferay/portal/search/tuning/synonyms/web/internal/portlet/action/EditSynonymSetsMVCActionCommandTest.java:121]: @LisztLiszt This variable is not needed either.

**Review by dantewang — CHANGES_REQUESTED** [review]
@LisztLiszt Please see my comment and resend. For names of variables you introduced, please make the change in your original commit; for variable inline, and other things, you can add SF commits.

**Pull request closed without merging**